external-crates/move/crates/move-package-alt/src/dependency/dependency_set.rs

```diff
@@ -0,0 +1,105 @@
+//! Conveniences for managing the entire collection of dependencies (including overrides) for a
+//! package
+use std::collections::BTreeMap;
+
+use crate::package::{EnvironmentName, PackageName};
+
+/// A set of default dependencies and dep overrides. Within each environment, package names are
+/// unique
+#[derive(Clone)]
+pub struct DependencySet<T> {
+    inner: BTreeMap<Option<EnvironmentName>, BTreeMap<PackageName, T>>,
+}
+
+impl<T> DependencySet<T> {
+    /// Return an empty set of dependencies
+    pub fn new() -> Self {
+        Self {
+            inner: BTreeMap::new(),
+        }
+    }
+
+    /// Combine all elements of [sets] into one. Any duplicate entries (with the same environment
+    /// and package name) are silently dropped.
+    pub fn merge(sets: impl IntoIterator<Item = Self>) -> Self {
+        // Note: we could use std::iter::flatten here, but implementing IntoIterator::IteratorType
```

Comment:
Not sure if there's a better way to do this

Reply:
I don't see how it's related - this is closer to flatten because it condenses an iterator of DependencySets into a single DependencySet.